src/lib/index/shared/redis.js

```diff
@@ -81,38 +78,56 @@ const
 
 		if (!redisInstance.subscribingConnection) {
 			redisInstance.subscribingConnection = redis.createClient(config);
-			redisInstance.subscribingConnection.on('messageBuffer', multiplexingHandler(config));
 		}
 
 		return redisInstance.subscribingConnection;
 	},
 
 	publishMessage = (channel, buffer) => (connection) => {
-		connection.publish(channel, buffer);
+		connection.lpush(channel, buffer);
 		return true;
 	},
 
 	subscribeHandler = (channel, handler, config) => (connection) => {
 
-		const redisInstance = getOrCreateRedisInstance(config);
+		const processMessage = () => {
 
-		connection.subscribe(channel);
-		redisInstance.subscribersByChannel.set(channel, handler);
+			connection.blpop(channel, 0, (err, data) => {
+				if (err) {
+					throw err;
+				}
+				handler(data[1]);
```

Comment:
What's the significance of data[1]? Maybe worth an explanatory comment.

Reply:
Good point. It is a rather interesting signature, the channel is the first entry in the array and the data the second.